src/main/java/mezz/jei/startup/ClientLifecycleHandler.java

```diff
@@ -139,4 +147,27 @@ private void onRecipesLoaded() {
 			);
 		}
 	}
+
+	private enum ServerType {
+		// Three cases, since there's no such thing as a vanilla integrated server
+		INTEGRATED(false, true, RecipesUpdatedEvent.class),
+		VANILLA_REMOTE(true, false, TagsUpdatedEvent.VanillaTagTypes.class),
+		MODDED_REMOTE(false, false, TagsUpdatedEvent.CustomTagTypes.class);
+
+		public final boolean isVanilla, isIntegrated;
+		public final Class<? extends Event> listenerClass;
+
+		private ServerType(boolean isVanilla, boolean isIntegrated, Class<? extends Event> listenerClass) {
+			this.isVanilla = isVanilla;
+			this.isIntegrated = isIntegrated;
+			this.listenerClass = listenerClass;
+		}
+
+		public boolean connected() {
+			ClientPlayNetHandler connection = Minecraft.getInstance().getConnection();
+			boolean isVanilla = connection != null && NetworkHooks.isVanillaConnection(connection.getNetworkManager());
```

Comment:
It probably doesn't really matter because most people should be on 1.16.5 now and at the very least 1.16.4, but realistically this PR due to using this helper method should have bumped the min forge version from 1.16.3-34.0.3 to 1.16.3-34.0.18